rebar.config

```diff
@@ -1,26 +1,59 @@
-    {erl_opts, [debug_info,
-                warn_missing_spec,
-                {i, ["include"]},
-                {src_dirs, ["src", "scenarios"]},
-                {parse_transform, lager_transform}]}.
+{erl_opts, [debug_info,
+            warn_missing_spec,
+            {i, ["include"]},
+            {src_dirs, ["src", "scenarios"]},
+            {parse_transform, lager_transform}]}.
 
-    {eunit_compile_opts, [nowarn_missing_spec]}.
+{eunit_compile_opts, [nowarn_missing_spec]}.
 
-    {require_otp_vsn, "^1[789]"}.
+{plugins, [pc,
+           rebar3_proper
+          ]}.
 
-    {deps, [
-            {lager, ".*", {git, "git://github.com/basho/lager.git", "3.2.1"}},
-            {jiffy, ".*", {git, "https://github.com/davisp/jiffy.git", "0.14.7"}},
-            {trails, ".*", {git, "https://github.com/inaka/cowboy-trails.git", {tag, "0.1.1"}}},
-            {cowboy_swagger, ".*", {git, "http://github.com/inaka/cowboy-swagger.git", {tag, "1.0.3"}}},
-            {escalus, ".*", {git, "git://github.com/esl/escalus.git", {tag, "3.0.0"}}},
-            {usec, ".*", {git, "git://github.com/esl/usec.git", {branch, "master"}}},
-            {exometer, ".*", {git, "git://github.com/Feuerlabs/exometer.git", "7a7bd8d2b52de4d90f65aa3f6044b0e988319b9e"}},
-            {lhttpc, ".*", {git, "git://github.com/esl/lhttpc.git", {branch, "otp-17-compat"}}},
-            {mochijson2, ".*", {git, "git://github.com/bjnortier/mochijson2.git", {branch, "master"}}},
-            {proper, ".*", {git, "git://github.com/manopapad/proper.git", {branch, "master"}}},
-            {recon, ".*", {git, "https://github.com/ferd/recon.git", "2.2.1"}},
-            {cowboy, ".*", {git, "https://github.com/ninenines/cowboy.git", "1.0.4"}},
-            {fusco, ".*", {git, "https://github.com/esl/fusco.git"}}
-           ]}.
+{require_otp_vsn, "^1[789]"}.
 
+{deps, [
+        {lager, "3.2.1"},
```

Comment:
Hi @michalwski ,
It is more about taking deps from the hex.pm instead of github. It'd be good to tag the current version(and actually introducing tagging, at the moment I can see none, apart from the magic 0.9 version, but it is not tagged), before pushing some breaking changes in the future. OFC If you insist I can restore github deps. Additionally proper also have to be taken back from the test profile, assuming that the profiles are not supported in r2

It is good that you mentioned the rebar2 compiled project. What is a strategy for developing custom scenarios that need to be closed sources? We are going to have some and putting them in the scenarios dir makes back porting harder. We need to be careful to commit only propertiary stuff, then "open" ones and use cherry-pick to move it to your repo. How do you handle it? You said that you used is as a dep, does it help with the custom scenarios case? 

The biggest drawback of "depping" I can see is that we will lose access to the make deploy commands. On the other hand maybe it is not a best idea to keep ansible stuff as a part of amoc repo. How do you solve it?

The other option I can see is renaming `scenarios` to `examples` or  `sample_scenarios` thanks to that we can clone our custom scenarios dir without touching core of amoc. Makefile could fallback to copy examples to the scenarios dir when the scenarios dir is not present, just to keep things going by default. It still needs a manual work(clone scenario repo), but It is better that current situation. 

All in all if you know ideal approach let me know!

To advertise ditching rebar2 compatibility - developer preview :) : https://github.com/ppikula/amoc_templates BTW I'm open to transfer ownership of this plugin to ESL when it is done. I'm not interested in maintaining it.